src/backend/core/services/quota.py

```
@@ -0,0 +1,318 @@
+class RecipientQuotaService:
+    """Service for managing recipient quotas in Redis."""
+
+    def __init__(self):
+        """Initialize service (Redis connection is lazy-loaded)."""
+        self._redis = None
+
+    @property
+    def redis(self):
+        """Lazy-load Redis connection."""
+        if self._redis is None:
+            from django.conf import settings
```

Ruff (0.14.10)

`import` should be at the top-level of a file

(PLC0415)